### DISCUSSION TITLE
Items scroll beyond container StyledScrollInventoryContainer

### DISCUSSION THREAD

**Discussion by ninagw** [neuefische/questions#829] [Web] [answered]
Hey,

I am currently working on my capstone project (Pocket Pets App - [GitHub-Link to project](https://github.com/Krischan-Klug/pocket-pets)), which displays a list of pet "beds" on the edit page (= `/edit/[id].js`). 

**Problem:** As soon as the max-width (300px) of the container is reached the items should scroll horizontally. Unfortunately, the items can be scrolled to the edge of the display width and do not end at the edges of the container (StyledScrollInventoryContainer). So the first item is always cut and is not shown properly.

<img width="362" alt="Bildschirm­foto 2024-04-23 um 16 34 38" src="https://github.com/neuefische/questions/assets/150810914/3f977e3e-651e-4bdf-af05-93544ff64745">

I tried to fix this problem by defining the left margin of the first item:

`pages/edit/[id].js`
```
const StyledScrollInventoryContainer = styled(StyledInventoryContainer)`
  width: 300px;
  height: 100px;
  overflow-x: auto;
  flex-wrap: nowrap;
  button:first-child {
    margin-left: calc((100vw - 300px) / 2);
  }
`;
```

`components/StyledComponents/StyledInventoryContainer.js`
```
const StyledInventoryContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  gap: 10px;
  margin: 10px 0;
`;
```

[My feature/refactor branch is accessible via this link.](https://github.com/Krischan-Klug/pocket-pets/tree/feature/pet-clothing-system-1)

Thank you for your help!

**Comment by JessicaLoers** [comment, marked as answer]
**Hello @ninagw** ✨

It looks like you're trying to control the **overflow behaviour** of a list of items in your styled container so that they scroll horizontally without any item getting cut off at the edges. Your current solution tries to adjust the margin of the first child to prevent it from being cut off, but there's a simpler and more effective way to handle this.

Let's  try to simplify the overflow and scrolling behaviour. 


```js
const StyledScrollInventoryContainer = styled.div`
  width: 300px; // Fixed width of the scrollable container
  overflow-x: auto;  // Enables horizontal scrolling
  display: flex;  // Use flexbox to align children inline
`;

const StyledInventoryWrapper = styled.div`
  display: flex;
  flex-wrap: nowrap;  // Prevents wrapping of children, forcing a single line
  gap: 10px;  //  Ensures spacing between items
  min-width: 0;  // // Fixes flexbox issues with overflow
`;
```

**Reply by ninagw** [reply]
Thank you!! 🙌
